update doc about Helm schematic workload (#1198)
oam-dev/kubevela: docs/en/helm/basic.md

# Use Helm chart as schematic module

> Here is an example of how to use Helm chart as workload capability module.

Reviewer: ```suggestion
Here is an example of how to use Helm chart as workload schematic module.
```

metadata:
  name: webapp-chart
  annotations:
    definition.oam.dev/description: helm chart for webapp
spec:
  definitionRef:
    name: deployments.apps
    version: v1
  schematic:
    helm:
      release:
        chart:
          spec:
            chart: "podinfo"
            version: "5.1.4"
      repository:
        url: "http://oam.dev/catalog/"
```

Just like using CUE as schematic module, we also have some rules and contracts to use helm chart as schematic module.

- `.spec.definitionRef` is required to indicate the main workload(Group/Verison/Kind) in your Helm chart.
Only one workload allowed in one helm chart.
For example, in our sample chart, the core workload is `deployments.apps/v1`, other resources will also be deployed but mechanism of KubeVela won't work for them.
- `.spec.schematic.helm` contains information of Helm release & repository.

Specifically, the definition follows the APIs from `fluxcd/flux2`, [HelmReleaseSpec](https://github.com/fluxcd/helm-controller/blob/main/docs/api/helmrelease.md) and [HelmRepositorySpec](https://github.com/fluxcd/source-controller/blob/main/docs/api/source.md#source.toolkit.fluxcd.io/v1beta1.HelmRepository).

> However, the fields shown in the sample are almost enough to describe a Helm chart release and its repository.

Reviewer: ```suggestion
In a word, just like the fields shown in the sample, the helm schematic module describes a Helm chart release and its repository.
```

## Create an Application using the helm based WorkloadDefinition

Here is an example `Application`.

```yaml
apiVersion: core.oam.dev/v1alpha2
kind: Application
metadata:
  name: myapp
  namespace: default
spec:
  components:
    - name: demo-podinfo 
      type: webapp-chart 
      settings: 
        image:
          tag: "5.1.2"
```
Helm module workload will use data in `settings` as [Helm chart values](https://github.com/captainroy-hy/podinfo/blob/master/charts/podinfo/values.yaml).
Currently, you can learn the schema of settings by reading the `README.md` of the Helm chart, and the schema are totally align with [`values.yaml`](https://github.com/captainroy-hy/podinfo/blob/master/charts/podinfo/values.yaml) of the chart.

> We will integerate with the [openapi-v3-json-schema automatically generation](https://kubevela.io/#/en/platform-engineers/openapi-v3-json-schema.md) soon.

Reviewer: ```suggestion
We will integerate the values and generate it's [openapi-v3-json-schema](https://kubevela.io/#/en/platform-engineers/openapi-v3-json-schema.md) soon.
```
